magicparse/schema.py

```diff
@@ -44,7 +44,20 @@ def register(cls, schema: "Schema") -> None:
 
         cls.registry[schema.key()] = schema
 
+    
     def parse(self, data: Union[bytes, BytesIO]) -> Tuple[List[dict], List[dict]]:
+        items = []
+        errors = []
+
+        for item, rowErrors in self.stream_parse(data):
+            if(rowErrors):
+                errors.extend(rowErrors)
+            else:
+                items.append(item)
+        
+        return items, errors
+    
+    def stream_parse(self, data: Union[bytes, BytesIO]) -> Iterable[Tuple[dict, dict]]:
```

Comment:
Pour info, il n'est plus nécessaire d’hériter des types de `typing` pour les collections, et il existe une syntaxe dédiée à l'union). le type dict n'est pas non plus complet, mais ça devient un poil complexe au niveau refacto, on peut utiliser Any pour la valeur, la clef par contre devrait tjrs etre des `str`, non ?
On pourrait réécrire le type en :
```
def stream_parse(self, data: bytes | BytesIO) -> Iterable[tuple[dict[str, Any], dict[str, Any]]]:
    ...
```

Reply:
Je suis d'accord avec toi Mathias !
Mais pour faire ça il faut d'abord bump la version de Python du projet dans le `Dockerfile` (passer de 3.9 à 3.11) et faire pareil dans `pyproject.toml`.
C'est pas un gros changement et de toute façon le seul service qui consomme magicparse est external-is qui est déjà en Python 3.11

Reply:
On peut faire ça dans une autre PR aussi, c'est pas directement lié à celle ci et ce n'est pas bloquant.